python/ml4ir/base/config/parse_args.py

```diff
@@ -299,6 +336,34 @@ def define_args(self):
     def set_default_args(self):
         pass
 
+    def parse_args(self, args: List[str]) -> Namespace:
+        """
+        Parse command line arguments passed as a list of strings.
+        Additionally, handles dynamic arguments for feature_config
+        and model_config with prefixes feature_config. and model_config.
+        respectively
+
+        Parameters
+        ----------
+        args: list of str
+            List of command line args to be parsed
+
+        Returns
+        -------
+        Namespace
+            Namespace object obtained by parsing the input
+        """
+        dynamic_args = self.parse_known_args(args)[1]
+
+        for i in range(int(len(dynamic_args) / 2)):
```

Comment:
this is not clear why we are doing it.. I mean the `range(int(len(dynamic_args) / 2))`
it took me time. 
You could: 
```
args, unknownargs = parser.parse_known_args(args)

# Unknown args, when exist, are a list that consists of pairs of key values. 
# args, unknownargs = parser.parse_known_args('-knowwn bar -unknown test'.split())
# returns unknownargs=['-unknown', 'test']
# Below, we iterate in pairs: 
for key, value in zip(dynamic_args[::2], dynamic_args[1::2]):
   ...
```